### PR TITLE
feat: incentive module msg and query framework

### PR DESCRIPTION
**Pull request by toteki**
## Description

Scaffolds out the keeper functions and command line interface for incentive module queries and messages, without implementing logic.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/umee/incentive/v1/query.proto [hunk 1]
```diff
@@ -95,6 +95,7 @@ message QueryPendingRewardsResponse {
 // QueryBonded defines the request structure for the Bonded gRPC service handler.
 message QueryBonded {
   string address = 1;
+  string denom   = 2;
```
Comment (robert-zaremba) [inline proto/umee/incentive/v1/query.proto:98]: shouldn't this be optional? if yes, then mark it optional and update the comment.

File: x/incentive/keeper/grpc_query.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	program := incentive.IncentiveProgram{}
+
+	resp := &incentive.QueryIncentiveProgramResponse{
+		Program: program,
+	}
+
+	return resp, incentive.ErrNotImplemented
```
Comment (robert-zaremba) [inline x/incentive/keeper/grpc_query.go:52]: let's remove the `incentive.ErrNotImplemented` error and use one of the standard grpc `status` codes or `sdkerrors`.

**Review by robert-zaremba — APPROVED** [review]
Looks good. Mock implementation of the generated interfaces.
One nit only:
- [ ] let's remove the custom `NotImplemented` and use one of the standard status codes or sdkerrors instead.

File: proto/umee/incentive/v1/query.proto (hunk 1 above)
Comment (toteki) [reply proto/umee/incentive/v1/query.proto:98]: In protobuf, `optional` is incompatible with `string` (but the string can be empty).

File: x/incentive/keeper/grpc_query.go (hunk 2 above)
Comment (toteki) [reply x/incentive/keeper/grpc_query.go:52]: I'll remove that error in the next few PRs (it's only a "TODO" type error during development)

**Pull request merged**